src/scenario_tests.rs

```diff
@@ -871,11 +885,77 @@ mod dst_tests {
             "Merge if we detect our section needs merging on CheckElder",
             &initial_state,
             &[
-                TestEvent::SetMergeNeeded(true).to_event(),
+                TestEvent::SetMergeNeeded(MergeInfo(OTHER_SECTION_INFO)).to_event(),
+                ParsecVote::CheckElder.to_event(),
+            ],
+            &AssertState {
+                action_our_events: vec![Rpc::Merge(OTHER_SECTION_INFO).to_event()],
+            },
+        );
+    }
+
+    #[test]
+    fn parsec_merge_sibling() {
+        let initial_state = arrange_initial_state(
+            &initial_state_old_elders(),
+            &[
+                TestEvent::SetMergeNeeded(MergeInfo(OTHER_SECTION_INFO)).to_event(),
```

Comment:
I don't really think we should pass the section info here:
We are merging, We know what our SectionInfo is, and that is what we will send in the RPC.
A bool is enough.